packages/core/test/errors.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const errors = require('../src/errors');
+const createAppRequestClient = require('../src/tools/create-app-request-client');
+const createInput = require('../src/tools/create-input');
+
+describe('errors', () => {
+  it('should name errors', () => {
+    const error = new errors.CheckError('Check Mate!');
+
+    error.should.instanceOf(errors.CheckError);
+    error.name.should.eql('CheckError');
+    error.message.should.eql('Check Mate!');
+  });
+
+  describe('AppError', () => {
+    it('should stringify arguments', () => {
+      const error = new errors.Error('My Message', 'MyCode', 400);
+
+      error.should.instanceOf(errors.Error);
+      error.name.should.eql('AppError');
+      error.message.should.eql(
+        '{"message":"My Message","code":"MyCode","status":400}'
+      );
+    });
+  });
+
+  describe('ResponseError', () => {
+    it('should stringify arguments', async () => {
+      const testLogger = () => Promise.resolve({});
+      const input = createInput({}, {}, testLogger);
```

Comment:
Personal learning question - what is going on here? I'm not sure why we're resolving a promise with an empty object or what createInput is doing for us.

Reply:
Good question - I copy pasted this from places like https://github.com/zapier/zapier-platform/blob/f3e81cab5c3fbcc68e4b0954de7c4c5edf9f2dd3/packages/core/test/tools/file-stasher.js#L17

Reply:
nothin fancy, that's calling this function: https://github.com/zapier/zapier-platform/blob/864229ce495d0a9b6244a78910f173ba696af55d/packages/core/src/tools/create-input.js#L12

In production, we call it here: https://github.com/zapier/zapier-platform/blob/864229ce495d0a9b6244a78910f173ba696af55d/packages/core/src/tools/create-lambda-handler.js#L208-L209